smartsim/entity/model.py

```diff
@@ -73,19 +83,20 @@ def __init__(
                                be added to run_settings
         :param batch_settings: Launcher settings for running the individual
                                application as a batch job
+        :param files: Files to have available to the application
         """
         super().__init__(name, str(path), run_settings)
         self.exe = [expand_exe_path(exe)]
         # self.exe = [exe] if run_settings.container else [expand_exe_path(exe)]
         self.exe_args = exe_args or []
-        self.params = params or {}
+        self.params = params.copy() if params else {}
```

Comment:
why shallow copy of params here? and not deep? but deep for files?

Reply:
Strings are immutable in python, a shallow copying is container of immutable objects is functionally equivalent to making deep copy because in order to change any of elements in the container it must be replaced wholesale!

 See [here](https://github.com/CrayLabs/SmartSim/pull/605#discussion_r1664601477) for my decision process for what I chose to deep/shallow/not copy, but I could be very easily convinced to make this a deep copy just for consistencies stake if you'd like!